pkg/webhook/statefulset/statefulset.go

```diff
@@ -110,12 +110,11 @@ func (sc *StatefulSetAdmissionControl) AdmitStatefulSets(ar *admission.Admission
 		return util.ARFail(err)
 	}
 
-	setPartition := *(stsPartition)
-	if setPartition > 0 && setPartition <= int32(partition) {
+	if stsPartition != nil && *stsPartition > 0 && *stsPartition <= int32(partition) {
 		klog.Infof("statefulset %s/%s has been protect by partition %s annotations", namespace, name, partitionStr)
 		return util.ARFail(errors.New("protect by partition annotation"))
 	}
-	klog.Infof("admit statefulset %s/%s update partition to %d, protect partition is %d", namespace, name, setPartition, partition)
+	klog.Infof("admit statefulset %s/%s update partition to %d, protect partition is %d", namespace, name, *stsPartition, partition)
```

Comment:
`stsPartition` is possible to be nil here.

Reply:
updated